TELCODOCS-2043 - CNF-13766 - unnumbered BGP peering (#89663)
openshift/openshift-docs: modules/nw-metallb-frr-k8s-configuration-crd.adoc

    foo: "bar"
----

[id="nw-metallb-frrconfiguration-crd-interface_{context}"]
== The interface field

> You can use the  `interface` field to dynamically discover IPv6 neighbors and create a direct connection without the need for IP addresses. 

Reviewer: I would avoid using the "dynamic"  word, and use the "unnumbered BGP" which a better reference for searching  the web, and is more occurate as we  do BGP over intefaces which do not have IP address (like unnumbered stands for  interface without ip address)

Author: Thanks. I've updated.

For example:

.Example FRRConfiguration CR
[source,yaml]
----
apiVersion: frrk8s.metallb.io/v1beta1
kind: FRRConfiguration
metadata:
  name: test
  namespace: frr-k8s-system
spec:
  bgp:
    bfdProfiles:
    - echoMode: false
      name: simple

>       passiveMode: false

Reviewer: can go awaw

    routers:
    - asn: 64512
      neighbors:
      - asn: 64512
        bfdProfile: simple
        disableMP: false
        interface: net10 <1>

>         passwordSecret: {}

Author: gone!

    foo: "bar"
----

[id="nw-metallb-frrconfiguration-crd-interface_{context}"]
== The interface field

include::snippets/technology-preview.adoc[]

You can use the  `interface` field to configure unnumbered BGP peering.
For example:

> .Example FRRConfiguration CR

Reviewer: Why in frrconfig we have a an example and not in the bgppeer?

    foo: "bar"
----

[id="nw-metallb-frrconfiguration-crd-interface_{context}"]
== The interface field

include::snippets/technology-preview.adoc[]

> You can use the `interface` field to configure unnumbered BGP peering.
> For example:

Reviewer 2: ```suggestion
You can use the `interface` field to configure unnumbered BGP peering by using the following example configuration:
```

> .Example FRRConfiguration CR

Reviewer 2: ```suggestion
.Example `FRRConfiguration` CR
```

[source,yaml]
----
apiVersion: frrk8s.metallb.io/v1beta1
kind: FRRConfiguration
metadata:
  name: test
  namespace: frr-k8s-system
spec:
  bgp:
    bfdProfiles:
    - echoMode: false
      name: simple
      passiveMode: false
    routers:
    - asn: 64512
      neighbors:
      - asn: 64512
        bfdProfile: simple
        disableMP: false
        interface: net10 <1>
        port: 179
        toAdvertise:
          allowed:
            mode: filtered
            prefixes:
            - 5.5.5.5/32
        toReceive:
          allowed:
            mode: filtered
      prefixes:
      - 5.5.5.5/32
----
<1> Activates unnumbered BGP peering.

[NOTE]
====

> To use the `interface` field, there must be a point-to-point, layer 2 connection between the two BGP peers.

Reviewer 2: ```suggestion
To use the `interface` field, you must establish a point-to-point, layer 2 connection between the two BGP peers.
```

|`spec.bgp.routers.neighbors.address`
|`string`
|Specifies the IP address to establish the session with.
If you use this field, you cannot specify a value in the `spec.bgp.routers.neighbors.interface` field.

|`spec.bgp.routers.neighbors.interface`
|`string`
|Specifies the interface name to use when establishing a session. 

> Use this field to dynamically discover IPv6 neighbors and create a direct connection without the need for IP addresses. 

Reviewer: still using dynamically

Author: d'Oh!

|`spec.bgp.routers.neighbors.address`
|`string`
|Specifies the IP address to establish the session with.
If you use this field, you cannot specify a value in the `spec.bgp.routers.neighbors.interface` field.

|`spec.bgp.routers.neighbors.interface`
|`string`
|Specifies the interface name to use when establishing a session. 
Use this field to configure unnumbered BGP peering.
There must be a point-to-point, layer 2 connection between the two BGP peers.
You can use unnumbered BGP peering with IPv4, IPv6, or dual-stack, but you must enable IPv6 RAs (Router Advertisements).
Each interface is limited to one BGP connection.
The `spec.bgp.routers.neighbors.interface` field is a Technology Preview feature only.

> For more information about the support scope of Red Hat Technology Preview features, see link:https://access.redhat.com/support/offerings/techpreview/[Technology Preview Features Support Scope].

Reviewer 2: ```suggestion
For more information about the support scope of Red{nbsp}Hat Technology Preview features, see link:https://access.redhat.com/support/offerings/techpreview/[Technology Preview Features Support Scope].
```
